v2/internal/controllers/generic_controller.go

```diff
@@ -184,86 +192,69 @@ func register(
-const NamespaceAnnotation = "serviceoperator.azure.com/operator-namespace"
-
 // Reconcile will take state in K8s and apply it to Azure
 func (gr *GenericReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
-	obj, err := gr.KubeClient.GetObjectOrDefault(ctx, req.NamespacedName, gr.GVK)
+	metaObj, err := gr.getObjectToReconcile(ctx, req)
 	if err != nil {
 		return ctrl.Result{}, err
 	}
-
-	if obj == nil {
+	if metaObj == nil {
 		// This means that the resource doesn't exist
 		return ctrl.Result{}, nil
 	}
 
-	// Always operate on a copy rather than the object from the client, as per
-	// https://github.com/kubernetes/community/blob/master/contributors/devel/sig-api-machinery/controllers.md, which says:
-	// Never mutate original objects! Caches are shared across controllers, this means that if you mutate your "copy"
-	// (actually a reference or shallow copy) of an object, you'll mess up other controllers (not just your own).
-	obj = obj.DeepCopyObject().(client.Object)
-
-	// The Go type for the Kubernetes object must understand how to
-	// convert itself to/from the corresponding Azure types.
-	metaObj, ok := obj.(genruntime.MetaObject)
-	if !ok {
-		return ctrl.Result{}, errors.Errorf("object is not a genruntime.MetaObject, found type: %T", obj)
-	}
+	originalObj := metaObj.DeepCopyObject().(genruntime.MetaObject)
 
 	log := gr.LoggerFactory(metaObj).WithValues("name", req.Name, "namespace", req.Namespace)
-	log.V(Verbose).Info(
-		"Reconcile invoked",
-		"kind", fmt.Sprintf("%T", obj),
-		"resourceVersion", obj.GetResourceVersion(),
-		"generation", obj.GetGeneration())
-	reconcilers.LogObj(log, "reconciling resource", metaObj)
+	reconcilers.LogObj(log, Verbose, "Reconcile invoked", metaObj)
 
 	// Ensure the resource is tagged with the operator's namespace.
-	annotations := metaObj.GetAnnotations()
-	reconcilerNamespace := annotations[NamespaceAnnotation]
-	if reconcilerNamespace != gr.Config.PodNamespace && reconcilerNamespace != "" {
-		// We don't want to get into a fight with another operator -
-		// so if we see another operator already has this object leave
-		// it alone. This will do the right thing in the case of two
-		// operators trying to manage the same namespace. It makes
-		// moving objects between namespaces or changing which
-		// operator owns a namespace fiddlier (since you'd need to
-		// remove the annotation) but those operations are likely to
-		// be rare.
-		message := fmt.Sprintf("Operators in %q and %q are both configured to manage this resource", gr.Config.PodNamespace, reconcilerNamespace)
-		gr.Recorder.Event(obj, corev1.EventTypeWarning, "Overlap", message)
-		return ctrl.Result{}, nil
-	} else if reconcilerNamespace == "" && gr.Config.PodNamespace != "" {
-		genruntime.AddAnnotation(metaObj, NamespaceAnnotation, gr.Config.PodNamespace)
-		return ctrl.Result{Requeue: true}, gr.KubeClient.Update(ctx, obj)
+	ownershipResult, err := gr.takeOwnership(ctx, metaObj)
```

Comment:
`err` redeclared here.. might need to create `var ownershipResult Result` separately and assign to it here

Reply:
This isn't actually redeclaring err, see https://go.dev/ref/spec#Short_variable_declarations

It's reusing the existing `err`. There is a case on line 228 though where, because of its different scope, `err` is redefined and I need to fix it.